In [1]:
"""
Programmer: Justin Wayne Daily
Name of Program: DailyJustinW_Module2_CaseStudy
Description: The program asks for the last name and the first name of a student.  After the student name is acquired, 
the program asks for the student's GPA.  The program informs the user if the student has made the Honor Roll, the 
Honor Roll and the Dean's List, or neither the Honor Roll and the Dean's List based on the student's GPA.  
The program is looped so that the user can continuously enter student names and student GPA's.  The user quits by 
entering the QUIT value for the student's last name.

List of variables:
QUIT - the sentinel value for the main program loop.  Entering the value for the student's last name will stop the program.
HONOR_ROLL_GPA - constant that represents the minimum GPA for making the Honor Roll.
DEANS_LIST_GPA - constant that represents the minimum GPA for making the Dean's List.
studentLastName - user input for the student's last name.  Its value is used to enter the main loop of the program.
studentFirstName - user input for the student's first name.  Both the student first name and student last name are echoed 
    to the user in program output.
studentGPA - user input for the student's GPA.  The student's GPA is compared the minimum GPA for both the Honor Roll and 
    the Dean's List, to see if the student has made either the Honor Roll, the Honor Roll and Dean's List, or neither the 
    Honor Roll and Dean's List.
"""

#The constants for the loop sentinel value, the minimum Honor Roll GPA, and the minimum Dean's List GPA are initialized
QUIT = "ZZZ"
HONOR_ROLL_GPA = 3.25
DEANS_LIST_GPA = 3.5

#Accept user input for the student last name, which acts as the entrance value for the main loop
studentLastName = input("Please enter a student's name, or enter " + QUIT + " to quit: ")

#Check if user input for the student last name matches the sentinel value for loop entry
#Accept user input for the student first name
#Accept user input for the student GPA
#Use decision to check if the student GPA is equal to or above the minimum GPA for Honor Roll status
#Output Honor Roll status if minimum Honor Roll GPA is met or exceeded
#Use decision to check if the student GPA is equal to or above the minimum GPA for Dean's List status
#Output Dean's List status if minimum Dean's List GPA is met or exceeded
#The decision for checking Dean's List status is nested in the decision for checking Honor Roll status 
    #because qualifying for the Honor Roll is a logical requirement for qualifying for the Dean's List
#If student GPA is below the minimum Honor Roll GPA (and technically the minimum Dean's List GPA), output 
    #student status for making neither the Honor Roll or the Dean's List
#Accept user input for the student last name to either check the Honor Roll and Dean's List status of a new 
    #student or to exit the program
while studentLastName != QUIT:
    studentFirstName = input("Please enter the first name for " + studentLastName + ": ")
    studentGPA = float(input("Please enter the GPA for " + studentFirstName + " " + studentLastName + ": "))
    if studentGPA >= HONOR_ROLL_GPA:
        print(studentFirstName + " " + studentLastName + " has made the Honor Roll.")
        if studentGPA >= DEANS_LIST_GPA:
            print(studentFirstName + " " + studentLastName + " has also made the Dean's List.")
    else:
        print(studentFirstName + " " + studentLastName + " has not made either the Honor Roll or the Dean's List.")
    print("")
    studentLastName = input("Please enter another student's last name, or enter " + QUIT + " to quit: ")

#Output final message to user to let them know that the program has ended
print("The program is finished.")

Please enter a student's name, or enter ZZZ to quit: Anne
Please enter the first name for Anne: Beth
Please enter the GPA for Beth Anne: 3.51
Beth Anne has made the Honor Roll.
Beth Anne has also made the Dean's List.

Please enter another student's last name, or enter ZZZ to quit: Anne
Please enter the first name for Anne: Anne
Please enter the GPA for Anne Anne: 3.5
Anne Anne has made the Honor Roll.
Anne Anne has also made the Dean's List.

Please enter another student's last name, or enter ZZZ to quit: Beth
Please enter the first name for Beth: Anne
Please enter the GPA for Anne Beth: 3.49
Anne Beth has made the Honor Roll.

Please enter another student's last name, or enter ZZZ to quit: Mann
Please enter the first name for Mann: Jack
Please enter the GPA for Jack Mann: 3.26
Jack Mann has made the Honor Roll.

Please enter another student's last name, or enter ZZZ to quit: Mann
Please enter the first name for Mann: Mann
Please enter the GPA for Mann Mann: 3.25
Mann Mann has made th

In [3]:
"""
Programmer: Justin Wayne Daily
Name of Program: DailyJustinW_Module2_CaseStudy
Description: The program asks for the last name and the first name of a student.  After the student name is acquired, 
the program asks for the student's GPA.  The program informs the user if the student has made the Honor Roll, the 
Honor Roll and the Dean's List, or neither the Honor Roll and the Dean's List based on the student's GPA.  
The program is looped so that the user can continuously enter student names and student GPA's.  The user quits by 
entering the QUIT value for the student's last name.

List of variables:
QUIT - the sentinel value for the main program loop.  Entering the value for the student's last name will stop the program.
HONOR_ROLL_GPA - constant that represents the minimum GPA for making the Honor Roll.
DEANS_LIST_GPA - constant that represents the minimum GPA for making the Dean's List.
studentLastName - user input for the student's last name.  Its value is used to enter the main loop of the program.
studentFirstName - user input for the student's first name.  Both the student first name and student last name are echoed 
    to the user in program output.
studentGPA - user input for the student's GPA.  The student's GPA is compared the minimum GPA for both the Honor Roll and 
    the Dean's List, to see if the student has made either the Honor Roll, the Honor Roll and Dean's List, or neither the 
    Honor Roll and Dean's List.
"""

#The constants for the loop sentinel value, the minimum Honor Roll GPA, and the minimum Dean's List GPA are initialized
QUIT = "ZZZ"
HONOR_ROLL_GPA = 3.25
DEANS_LIST_GPA = 3.5

#Use "while True" loop for main program loop
#Accept user input for the student last name
#Use decision to check if student last name matches the sentinel value for the loop
#Break the loop if the user input for student last name matches the sentinel value for the loop
#Accept user input for the student first name
#Accept user input for the student GPA
#Use decision to check if the student GPA is equal to or above the minimum GPA for Honor Roll status
#Output Honor Roll status if minimum Honor Roll GPA is met or exceeded
#Use decision to check if the student GPA is equal to or above the minimum GPA for Dean's List status
#Output Dean's List status if minimum Dean's List GPA is met or exceeded
#The decision for checking Dean's List status is nested in the decision for checking Honor Roll status 
    #because qualifying for the Honor Roll is a logical requirement for qualifying for the Dean's List
#If student GPA is below the minimum Honor Roll GPA (and technically the minimum Dean's List GPA), output 
    #student status for making neither the Honor Roll or the Dean's List
#Accept user input for the student last name to either check the Honor Roll and Dean's List status of a new 
    #student or to exit the program
while True:
    studentLastName = input("Please enter a student's last name, or enter " + QUIT + " to quit: ")
    if studentLastName == QUIT:
        break
    studentFirstName = input("Please enter the first name for " + studentLastName + ": ")
    studentGPA = float(input("Please enter the GPA for " + studentFirstName + " " + studentLastName + ": "))
    if studentGPA >= HONOR_ROLL_GPA:
        print(studentFirstName + " " + studentLastName + " has made the Honor Roll.")
        if studentGPA >= DEANS_LIST_GPA:
            print(studentFirstName + " " + studentLastName + " has also made the Dean's List.")
    else:
        print(studentFirstName + " " + studentLastName + " has not made either the Honor Roll or the Dean's List.")
    print("")


#Output final message to user to let them know that the program has ended
print("The program is finished.")

Please enter a student's last name, or enter ZZZ to quit: Anne
Please enter the first name for Anne: Beth
Please enter the GPA for Beth Anne: 3.51
Beth Anne has made the Honor Roll.
Beth Anne has also made the Dean's List.

Please enter a student's last name, or enter ZZZ to quit: Anne
Please enter the first name for Anne: Anne
Please enter the GPA for Anne Anne: 3.5
Anne Anne has made the Honor Roll.
Anne Anne has also made the Dean's List.

Please enter a student's last name, or enter ZZZ to quit: Beth
Please enter the first name for Beth: Anne
Please enter the GPA for Anne Beth: 3.49
Anne Beth has made the Honor Roll.

Please enter a student's last name, or enter ZZZ to quit: Mann
Please enter the first name for Mann: Jack
Please enter the GPA for Jack Mann: 3.26
Jack Mann has made the Honor Roll.

Please enter a student's last name, or enter ZZZ to quit: Mann
Please enter the first name for Mann: Mann
Please enter the GPA for Mann Mann: 3.25
Mann Mann has made the Honor Roll.

Plea

## Test Data

### Test One - Slightly above Dean's List GPA
>Please enter a student's name, or enter ZZZ to quit: Anne<br>
>Please enter the first name for Anne: Beth<br>
>Please enter the GPA for Beth Anne: 3.51<br>
>Beth Anne has made the Honor Roll.<br>
>Beth Anne has also made the Dean's List.<br>

### Test Two - Equal to Dean's List GPA
>Please enter another student's last name, or enter ZZZ to quit: Anne<br>
>Please enter the first name for Anne: Anne<br>
>Please enter the GPA for Anne Anne: 3.5<br>
>Anne Anne has made the Honor Roll.<br>
>Anne Anne has also made the Dean's List.<br>

### Test Three - Slightly below Dean's List GPA
>Please enter another student's last name, or enter ZZZ to quit: Beth<br>
>Please enter the first name for Beth: Anne<br>
>Please enter the GPA for Anne Beth: 3.49<br>
>Anne Beth has made the Honor Roll.<br>

### Test Four - Slightly above Honor Roll GPA
>Please enter another student's last name, or enter ZZZ to quit: Mann<br>
>Please enter the first name for Mann: Jack<br>
>Please enter the GPA for Jack Mann: 3.26<br>
>Jack Mann has made the Honor Roll.<br>

### Test Five - Equal to Honor Roll GPA
>Please enter another student's last name, or enter ZZZ to quit: Mann<br>
>Please enter the first name for Mann: Mann<br>
>Please enter the GPA for Mann Mann: 3.25<br>
>Mann Mann has made the Honor Roll.<br>

### Test Six - Slightly below Honor Roll GPA
>Please enter another student's last name, or enter ZZZ to quit: Jack<br>
>Please enter the first name for Jack: Mann<br>
>Please enter the GPA for Mann Jack: 3.24<br>
>Mann Jack has not made either the Honor Roll or the Dean's List.<br>

### Test Seven - Exitng the program
>Please enter another student's last name, or enter ZZZ to quit: ZZZ<br>
>The program is finished.<br>